### DISCUSSION TITLE
Comparing InfoNCE between Behavior and Hybrid models

### DISCUSSION THREAD

**Discussion by BrainofZedZed** [AdaptiveMotorControlLab/CEBRA#165] [Q&A]
Hi CEBRA community,

I'm getting a lot of CEBRA -- thanks for releasing it to the community!

I have a variation of a [recent question](https://github.com/AdaptiveMotorControlLab/CEBRA/discussions/159). I'm interested in understanding the contributions of auxiliary variables to embeddings. I've used Behavior models to visualize the embeddings and compare InfoNCE loss with different variables contributing to the embeddings. I've also used Hybrid models to do the same. I see consistent patterns of loss between Behavior and Hybrid, but the value of the InfoNCE loss is scaled differently (between 5 and 6 for Behavior, 11 and 12 for Hybrid).  I have two questions:

- why is the InfoNCE loss higher for Hybrid models than Behavior models? Naively, I would expect the opposite.
- I notice in the demo notebooks comparing models trained with different variables, only Behavior models are used. Is there a reason one would not want to use Hybrid models for the same purpose?

**Comment by stes** [comment]
Hi @BrainofZedZed , this is most likely because of different batch sizes. Can you post a minimum example highlighting your issue? We can take it from there then.
